### ISSUE TITLE
Installing Version 3.1.0 and importing package results in a "Cannot read property 'fileName' of null" Error

### ISSUE THREAD

**Issue by DanielSandl** [fusioncharts/angular2-fusioncharts#99]
When installing version 3.1.0 and adding this line:

`import { FusionChartsModule } from 'angular-fusioncharts';`

It results in an error saying "Cannot read property 'fileName' of null." This is not a problem with our repo, as this even happened when following the steps in the [angular tutorial.](https://www.fusioncharts.com/dev/getting-started/angular/angular/your-first-chart-using-angular) In case anyone else gets this, the workaround is to downgrade to 3.0.4

**Comment by fellmirr** [comment]
Unfortunately, this was not solved with a downgrade for me. It simply gives a different error 
`Entry point '@angular/core' contains deep imports into (...)`

**Comment by TroniPM** [comment]
> When installing version 3.1.0 and adding this line:
> 
> `import { FusionChartsModule } from 'angular-fusioncharts';`
> 
> It results in an error saying "Cannot read property 'fileName' of null." This is not a problem with our repo, as this even happened when following the steps in the [angular tutorial.](https://www.fusioncharts.com/dev/getting-started/angular/angular/your-first-chart-using-angular) In case anyone else gets this, the workaround is to downgrade to 3.0.4

**This worked for me**. I'm using ```angular 11.0.5```.

**Comment by jerone** [comment]
Same issue here. This same thing happens when you run `ngcc` after installing.

Downgrading worked.

**Comment by AyanBhadury** [comment]
Please try with angular-fusioncharts: 3.0.4

**Comment by omanzelli** [comment]
Same issue and fixed downgrading the version

**Comment by serhiihordiichuk** [comment]
Please check again and remove **^** from npm version
> "angular-fusioncharts": "3.0.4",

**Comment by SpiderPork** [comment]
Same problem for me.

**Comment by AyanBhadury** [comment]
Here is a working demo: https://github.com/AyanBhadury/angular-fusioncharts-demo

**Comment by SpiderPork** [comment]
> 
> 
> Here is a working demo: https://github.com/AyanBhadury/angular-fusioncharts-demo

With angular-fusioncharts 3.0.4 and Angular 10.1.6.
Did you try it with Angular 11.1.1 and angular-fusioncharts 3.1.0?

**Comment by AyanBhadury** [comment]
As mentioned before use angular-fusioncharts: 3.0.4

3.1.0 has an issue with ngcc compilation

**Comment by SpiderPork** [comment]
> 
> 
> As mentioned before use angular-fusioncharts: 3.0.4
> 
> 3.1.0 has an issue with ngcc compilation

Is there an issue tracking ngcc compilation compatibility?

**Comment by AyanBhadury** [comment]
drop a mail to support@fusioncharts.com with your license details

> > As mentioned before use angular-fusioncharts: 3.0.4
> > 3.1.0 has an issue with ngcc compilation
> 
> Is there an issue tracking ngcc compilation compatibility?

**Comment by jguo1002** [comment]
> Please check again and remove **^** from npm version
> 
> > "angular-fusioncharts": "3.0.4",

Yes removing the caret works, otherwise same error above. I'm using `angular 11.12.14`.

**Comment by AyanBhadury** [comment]
ngCC compilation issue has been fixed, please upgrade to 3.2.0 : http://www.npmjs.com/package/angular-fusioncharts
@jerone @fellmirr @serhiihordiichuk @TroniPM @jguo1002

**Comment by jerone** [comment]
@AyanBhadury That's great news! 🎉

Where can we see the changes/changelog for the latest update?

The last commit on the [develop](https://github.com/fusioncharts/angular-fusioncharts/commits/develop) branch is from June.  
All [other branches](https://github.com/fusioncharts/angular-fusioncharts/branches) appear to be stale or bots.  
[Releases page](https://github.com/fusioncharts/angular-fusioncharts/releases) is missing version information since 3.1.  
There is no changelog file.  
There is no information on the [readme](https://github.com/fusioncharts/angular-fusioncharts#readme) in this repo or [NPM](https://www.npmjs.com/package/angular-fusioncharts).  
There is no information on the [documentation page](https://fusioncharts.github.io/angular-fusioncharts/#/ex1).

**Comment by AyanBhadury** [comment]
@jerone the documentation is not updated yet, I will try to sync the GitHub repo soon, rest assured you can download the latest version from npmjs, internally only we have updated a few build configurations, implementation wise there is no change hence no other extra configuration is required at the user's end.
